src/iuse_actor.cpp

```diff
@@ -1566,6 +1566,44 @@ long fireweapon_on_actor::use( player *p, item *it, bool t, const tripoint& ) co
     return it->type->charges_to_use();
 }
 
+void manualnoise_actor::load( JsonObject &obj )
+{
+    no_charges_message  = obj.get_string( "no_charges_message" );
+    use_message         = obj.get_string( "use_message" );
+    noise_message       = obj.get_string( "noise_message", "" );
+    noise               = obj.get_int( "noise", 0 );
+    moves               = obj.get_int( "moves", 0 );
+}
+
+iuse_actor *manualnoise_actor::clone() const
+{
+    return new manualnoise_actor( *this );
+}
+
+long manualnoise_actor::use( player *p, item *it, bool t, const tripoint& ) const
+{
+    if( t ) {
+        return 0;
+    }
+    if( it->charges <= 0 ) {
```

Comment:
Bicycle horns don't have charges, because their action (`iuse::horn_bicycle`) does not consume any charges and therefor they do not start with any charges (their JSON entry: initial charges, maximal charges, charges per use are all 0).

As you're going to combine the two iuse functions (one for a tool using charges and one for a tool not using charges), you should make sure the old horn still work as they used to work. Like testing for either `charges_to_use` returning 0 or, charges being large enough:

```C++
if( it->type->charges_to_use() != 0 && it->charges < it->type->charges_to_use() ) {
    return 0;
}
```